ui/pages/confirm-approve/confirm-approve.js

```diff
@@ -162,21 +132,30 @@ export default function ConfirmApprove() {
 
   const { iconUrl: siteImage = '' } = subjectMetadata[origin] || {};
 
-  const tokensText = `${Number(tokenAmount)} ${tokenSymbol}`;
-  const tokenBalance = tokenTrackerBalance
-    ? calcTokenAmount(tokenTrackerBalance, decimals).toString(10)
+  let tokensText;
+  if (assetStandard === ERC20) {
+    tokensText = `${Number(tokenAmount)} ${tokenSymbol}`;
+  } else if (assetStandard === ERC721 || assetStandard === ERC1155) {
+    tokensText = assetName;
+  }
+
+  const tokenBalance = userBalance
+    ? calcTokenAmount(userBalance, decimals).toString(10)
     : '';
   const customData = customPermissionAmount
-    ? getCustomTxParamsData(data, { customPermissionAmount, decimals })
+    ? getCustomTxParamsData(transactionData, {
+        customPermissionAmount,
+        decimals,
+      })
     : null;
 
-  return tokenSymbol === undefined ? (
+  return tokenSymbol === undefined && assetName === undefined ? (
     <Loading />
   ) : (
     <GasFeeContextProvider transaction={transaction}>
       <ConfirmTransactionBase
         toAddress={toAddress}
-        identiconAddress={tokenAddress}
+        identiconAddress={toAddress}
```

Comment:
the identicon we show on the approve screen should be for the address we are granting approval not the token for which we are granting approval:
<img width="358" alt="Screen Shot 2022-02-28 at 12 57 50 PM" src="https://user-images.githubusercontent.com/34557516/156050239-02248807-af86-4366-8433-13c84af3329e.png">